cpp/net/connection_pool.cc

```diff
@@ -90,12 +229,14 @@ unique_ptr<ConnectionPool::Connection> ConnectionPool::Get(const URL& url) {
 
   if (it == conns_.end() || it->second.empty()) {
     VLOG(1) << "new evhtp_connection for " << key.first << ":" << key.second;
-    unique_ptr<ConnectionPool::Connection> conn(
-        new Connection(base_->HttpConnectionNew(key.first, key.second),
-                       move(key)));
-    evhtp_set_hook(&conn->connection()->hooks, evhtp_hook_on_connection_fini,
+    unique_ptr<ConnectionPool::Connection> conn(new Connection(
+        url.Protocol() == "https"
+            ? base_->HttpsConnectionNew(key.first, key.second, ssl_ctx_.get())
```

Comment:
This here is blocking at the moment, which could be fixed in the future, but for now, calls to this method are on the libevent thread, where blocking is Very Bad (tm).

Keeping it for a follow up pull request, as discussed (my head most definitely has a bounded capacity!), but we'll have to work around this, otherwise untold horrors will get us later.